processor/transformprocessor/internal/metrics/func_convert_summary_count_val_to_sum.go

```diff
@@ -0,0 +1,62 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func convertSummaryCountValToSum(stringAggTemp string, monotonic bool) (common.ExprFunc, error) {
+	var aggTemp pmetric.MetricAggregationTemporality
+	switch stringAggTemp {
+	case "delta":
+		aggTemp = pmetric.MetricAggregationTemporalityDelta
+	case "cumulative":
+		aggTemp = pmetric.MetricAggregationTemporalityCumulative
+	default:
+		return nil, fmt.Errorf("unknown aggregation temporality: %s", stringAggTemp)
+	}
+	return func(ctx common.TransformContext) interface{} {
+		mtc, ok := ctx.(metricTransformContext)
+		if !ok {
+			return nil
+		}
+
+		metric := mtc.GetMetric()
+		if metric.DataType() != pmetric.MetricDataTypeSummary {
+			return nil
+		}
+
+		sumMetric := mtc.GetMetrics().AppendEmpty()
+		sumMetric.SetDescription(metric.Description())
```

Comment:
Will the description translate correctly?  Should the function take a "description" parameter so that users can set the description to be whatever they want?  Would be nice if it could be optional. Also applies to the other function.

Reply:
probably wouldn't be completely correct but figured it was better than not transferring anything. Again, is there support for optional params?

Reply:
Could probably be set with the set function later.  Please document all fields that are copied.

Reply:
Documented